apps/srp.c

```diff
@@ -205,30 +207,32 @@ static char *srp_verify_user(const char *user, const char *srp_verifier,
                              char *srp_usersalt, const char *g, const char *N,
                              const char *passin, BIO *bio, int verbose)
 {
-    char password[1024];
+    char password[1025];
     PW_CB_DATA cb_tmp;
     char *verifier = NULL;
     char *gNid = NULL;
+    int len;
 
     cb_tmp.prompt_info = user;
     cb_tmp.password = passin;
 
-    if (password_callback(password, 1024, 0, &cb_tmp) > 0) {
+    len = password_callback(password, sizeof(password)-1, 0, &cb_tmp);
+    if (len > 0) {
+        password[len] = 0;
         VERBOSE BIO_printf(bio,
-                           "Validating\n   user=\"%s\"\n srp_verifier=\"%s\"\n srp_usersalt=\"%s\"\n g=\"%s\"\n N=\"%s\"\n",
+			   "Validating\n   user=\"%s\"\n srp_verifier=\"%s\"\n srp_usersalt=\"%s\"\n g=\"%s\"\n N=\"%s\"\n",
```

Comment:
??? Original indentation was correct. What would be appropriate is to break this line to fit recommended width, for example:
```
VERBOSE BIO_printf(bio, "Validating\n   user=\"%s\"\n srp_verifier=\"%s\"\n"
                        " srp_usersalt=\"%s\"\n g=\"%s\"\n N=\"%s\"\n",
                   user, srp_verifier, srp_usersalt, g, N);
```
But then it would be appropriate to fix it even elsewhere. In other words just keep original indentation [here and now].